### DISCUSSION TITLE
Keyboard with Chinese spelling phonemes does not combine together when speaking

### DISCUSSION THREAD

**Discussion by su-aac** [UNC-Project-Open-AAC/OS-DPI#176] [Help]
Hello everyone!

I need some help with my keyboard display.

I made a keyboard with the Chinese spelling phonemes- "Zhuyin" phonetic system. However, when I entered the phonemes on the display bar and pressed "Speak", it spoke out each phoneme rather than spelling/combining them together. For example, in English, I type C-A-T, and it should say "cat" instead of single letters.
![image](https://user-images.githubusercontent.com/123387291/229259783-d92790e2-cf55-4aad-8709-3b7e355608f1.png)
Zhuyin keyboard looks like this (picture from website).
The phonemes on the right half of the keyboard 一ㄨㄩㄚㄛㄜㄝㄞㄟㄠㄡㄢㄣㄤㄥㄦ are vowels, and the others on the left half of the keyboard are consonants.

In my excel sheet, I have the Chinese phonemes in the "case" column named "lower" and English capital letters in the same column named "upper". But when I clicked on the "CAP" in the display, it did not change the page to uppercase English letters.

I copied the link here. Thank you.
https://unc-project-open-aac.github.io/OS-DPI/#Keyboard_Chinese 
My design file is here!
[Keyboard_Chinese.zip](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/11128233/Keyboard_Chinese.zip)

Sophie

**Comment by sofia-bg** [comment]
@hsiaotingsu would be interested in checking out your design - can you export the file and upload it? (File -> Export )
The url won't work because the data are all saved in your browser. 

Then in order to upload directly to discussions you have to save as a .zip file. I just rename from .osdpi to .zip after I export to my computer.

**Reply by su-aac** [reply]
@sofia-bg Thank you!! My design file is here!
[Keyboard_Chinese.zip](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/11128236/Keyboard_Chinese.zip)

**Comment by gbishop** [comment]
What Voice did you select? Your code appears correct to me but I have no idea what I should be hearing so it is hard to help. 

Grasping at straws, I'm thinking the choice of text to speech voice would make a huge difference in how it behaves.

**Reply by su-aac** [reply]
@gbishop Wow!! Changing the voice to Google did work! Thank you!

**Reply by gbishop** [reply]
Great! This would make a great note for the documentation. Our system knows nothing of language; in the future the NLP components will do that. For now it is simply moving symbols around. How the collection of symbols is interpreted by the text-to-speech engine is entirely determined by the Voice you choose.

**Reply by aetkie** [reply]
Noted this in the Wiki [here](https://github.com/UNC-Project-Open-AAC/OS-DPI/wiki/Speech).
<img width="608" alt="Screenshot 2023-04-03 at 2 29 39 PM" src="https://user-images.githubusercontent.com/113473933/229597924-e3e316b5-7099-4247-be06-a3ae5997e95d.png">

**Reply by aetkie** [reply]
@gbishop @hsiaotingsu @sofia-bg I am revisiting this topic for clarification. Can anyone provide steps on how they added the Google-Mandarin voice to their machine to make this UI work correctly?

**Reply by sofia-bg** [reply]
@aetkie it should just be a voice option in chrome, so no one should need to import a voice. rather they just need to go through the drop down menu until they find the right one

**Comment by gbishop** [comment]
Excellent!

**Comment by gbishop** [comment]
In my list of voices I see several that are identified only by some Kanji?
characters. I bet it is one of them.

**Reply by su-aac** [reply]
Thanks, Gary and Sofia! I found out that the "Shu-Yu" voice in the drop-down menu on Andrea's laptop works. Interestingly, my laptop, which I purchased in Taiwan, has different voice options in the drop-down menu. It looks like this:
![414243909_1093489601895013_6994895171198662125_n](https://github.com/UNC-Project-Open-AAC/OS-DPI/assets/123387291/bcebeb05-9ef5-4961-b566-683ef37bd64f)
